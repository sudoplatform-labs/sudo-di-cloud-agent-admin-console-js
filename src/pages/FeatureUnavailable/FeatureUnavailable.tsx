import React from 'react';
import styled from 'styled-components';
import { Description, Heading } from '../../components/charts/ChartHeading';
import { ConsoleCard } from '../../components/ConsoleCard';
import {
  CardsCol,
  CardsRow,
  PageHeading,
  Page,
} from '../../components/NavLayout';
import { Row, Col, Button } from 'antd';
import { Image, ImageName } from '../../components/Image';
import { theme } from '../../theme';

const StyledConsoleCard = styled(ConsoleCard)`
  .ant-card-body {
    padding: 36px 48px;
  }
`;

const StyledCol = styled(Col)`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const CTAHeading = styled(Heading)`
  margin-bottom: 8px;
  font-size: 20px;
  font-weight: 900;
  line-height: 1.2;
`;

const CTADescription = styled(Description)`
  margin-bottom: 0;
`;

const CTASubDescription = styled(Description)`
  color: ${theme.greys.chateau};
`;

const CTAButton = styled(Button)`
  max-width: 144px;
`;

const StyledImage = styled(Image)`
  height: auto;
  width: 100%;
`;

interface Props {
  pageHeading: string;
  ctaHeading: string;
  ctaDescription: string;
  ctaSubDescription: string;
  ctaLink: string;
  imageName: ImageName;
}

export const FeatureUnavailable: React.FC<Props> = (props) => {
  return (
    <Page>
      <PageHeading>{props.pageHeading}</PageHeading>
      <CardsRow>
        <CardsCol span={24}>
          <StyledConsoleCard>
            <Row gutter={24}>
              <StyledCol span="12">
                <CTAHeading>{props.ctaHeading}</CTAHeading>
                <CTADescription>{props.ctaDescription}</CTADescription>
                <CTASubDescription>{props.ctaSubDescription}</CTASubDescription>
                <CTAButton
                  type="primary"
                  shape="round"
                  href={props.ctaLink}
                  target="_blank"
                >
                  Learn More
                </CTAButton>
              </StyledCol>
              <StyledCol span="12">
                <StyledImage name={props.imageName} />
              </StyledCol>
            </Row>
          </StyledConsoleCard>
        </CardsCol>
      </CardsRow>
    </Page>
  );
};
