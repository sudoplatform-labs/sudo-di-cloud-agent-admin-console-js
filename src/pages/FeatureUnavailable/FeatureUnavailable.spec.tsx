import { shallow } from 'enzyme';
import React from 'react';
import { FeatureUnavailable } from './FeatureUnavailable';

describe('FeatureUnavailable', () => {
  it('should render correctly', () => {
    const component = shallow(
      <FeatureUnavailable
        pageHeading="Chris in Montana"
        ctaHeading="Chris is the name and desertion is the game"
        ctaDescription="Chris left me and I can't function..."
        ctaSubDescription="CHRISTOPHER!!!!  WHY!!!!!"
        ctaLink="www.never-leave-again.com"
        imageName="settings"
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
