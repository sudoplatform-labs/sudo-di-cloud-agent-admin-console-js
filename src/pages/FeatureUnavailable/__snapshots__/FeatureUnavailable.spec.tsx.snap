// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureUnavailable should render correctly 1`] = `
<styled.div>
  <styled.h1>
    Chris in Montana
  </styled.h1>
  <CardsRow>
    <Styled(Col)
      span={24}
    >
      <Styled(Styled(Card))>
        <Row
          gutter={24}
        >
          <Styled(Col)
            span="12"
          >
            <Styled(styled.h3)>
              Chris is the name and desertion is the game
            </Styled(styled.h3)>
            <Styled(styled.p)>
              Chris left me and I can't function...
            </Styled(styled.p)>
            <Styled(styled.p)>
              CHRISTOPHER!!!!  WHY!!!!!
            </Styled(styled.p)>
            <Styled(Button)
              href="www.never-leave-again.com"
              shape="round"
              target="_blank"
              type="primary"
            >
              Learn More
            </Styled(Button)>
          </Styled(Col)>
          <Styled(Col)
            span="12"
          >
            <Styled(Image)
              name="settings"
            />
          </Styled(Col)>
        </Row>
      </Styled(Styled(Card))>
    </Styled(Col)>
  </CardsRow>
</styled.div>
`;
